classes/test.php

```diff
@@ -411,6 +411,27 @@ public function setAssertionManager(test\assertion\manager $assertionManager = n
 			->setHandler('mockGenerator', function() use ($test) { return $test->getMockGenerator(); })
 			->setHandler('mockClass', function($class, $mockNamespace = null, $mockClass = null) use ($test) { $test->getMockGenerator()->generate($class, $mockNamespace, $mockClass); return $test; })
 			->setHandler('mockTestedClass', function($mockNamespace = null, $mockClass = null) use ($test) { $test->getMockGenerator()->generate($test->getTestedClassName(), $mockNamespace, $mockClass); return $test; })
+			->setHandler('newMockInstance', function($class, $mockNamespace = null, $mockClass = null, array $constructorArguments = null) use ($test)
+				{
+					$mockNamespace = trim($mockNamespace ?: $test->getMockGenerator()->getDefaultNamespace(), '\\');
+					$mockClass = trim($mockClass ?: $class, '\\');
+					$className = $mockNamespace . '\\' . $mockClass;
+
+					if (class_exists($className) === false)
+					{
+						$test->mockClass($class, $mockNamespace, $mockClass);
+					}
+
+					if ($constructorArguments !== null)
```

Comment:
yes at some places we use a reflectionClassFactory (for example in the constructor of this class) to deeply test code behavior but it was not required here: those tests are tightly coupled to implementation, I prefered to only check external behaviors